crypto/ex_data.c

```diff
@@ -358,11 +358,11 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         if (storage[i] && storage[i]->dup_func)
             storage[i]->dup_func(to, from, &ptr, i,
                                  storage[i]->argl, storage[i]->argp);
-        CRYPTO_set_ex_data(to, i, ptr);
+        ret &= CRYPTO_set_ex_data(to, i, ptr);
```

Comment:
It's currently unclear to me what the correct behaviour should be in case of failure, if this leaves it in a consistent state on failure or not.  At least the public and documented function EC_KEY_copy() is making use of it and just returns NULL on failure while we probably touched it's content.